python_modules/dagster/dagster/_core/execution/run_metrics_thread.py

```diff
@@ -172,7 +182,9 @@ def _capture_metrics(
     check.opt_inst_param(logger, "logger", logging.Logger)
 
     if not (container_metrics_enabled or python_metrics_enabled):
-        raise Exception("No metrics enabled")
+        if logger:
+            logger.debug("No metrics to capture, shutting down metrics capture thread")
```

Comment:
Does this get captured at the default log level? If a tag has been set indicating that we want metrics but no metrics are going to come, that seems at least info worthy if not warn? (definitely agree that we shouldn't kill the process for this though)